File: backend/src/main/java/com/NBE3_4_2_Team4/global/api/iamport/v1/account/IamportAccountService.java
package com.NBE3_4_2_Team4.global.api.iamport.v1.account;

import java.util.List;
import java.util.Optional;

import static com.NBE3_4_2_Team4.global.api.iamport.v1.account.IamportAccountRequestDto.*;
import static com.NBE3_4_2_Team4.global.api.iamport.v1.account.IamportAccountResponseDto.*;

public interface IamportAccountService {

    Optional<String> validateBankAccount(String accessToken, BankAccountValidator bankAccount);

    List<BankInfo> getBankCodes(String accessToken);

    Optional<BankInfo> findBankNameByBankCode(String accessToken, String bankCode);
}

File: backend/src/main/java/com/NBE3_4_2_Team4/global/api/iamport/v1/account/IamportAccountService.kt
package com.NBE3_4_2_Team4.global.api.iamport.v1.account

import com.NBE3_4_2_Team4.global.api.iamport.v1.account.IamportAccountRequestDto.BankAccountValidator
import com.NBE3_4_2_Team4.global.api.iamport.v1.account.IamportAccountResponseDto.BankInfo

interface IamportAccountService {

    fun validateBankAccount(
        accessToken: String,
        bankAccount: BankAccountValidator
    ) : String?

    fun getBankCodes(
        accessToken: String
    ) : List<BankInfo>

    fun findBankNameByBankCode(
        accessToken: String,
        bankCode: String
    ) : BankInfo?
}